docs/ipfs-cluster-guide.md

```diff
@@ -56,23 +56,91 @@ You can deploy cluster in the way which fits you best, as both ipfs and ipfs-clu
+
+Each section of the configuration file and the options in it depend on their associated component. We offer here a quick reference of the configuration format:
+
+```json
+{
+  "cluster": { // main cluster component configuration
+    "id": "QmZyXksFG3vmLdAnmkXreMVZvxc4sNi1u21VxbRdNa2S1b", // peer ID
+    "private_key": "<base64 representation of the key>",
+    "secret": "<32-bit hex encoded secret>",
+    "peers": [], // List of peers' multiaddresses
+    "bootstrap": [], // List of bootstrap peers' multiaddresses
+    "leave_on_shutdown": false, // Abandon consensus on exit
+    "listen_multiaddress": "/ip4/0.0.0.0/tcp/9096", // Cluster RPC listen
+    "state_sync_interval": "1m0s", // Time between state syncs
+    "ipfs_sync_interval": "2m10s", // Time between ipfs-state syncs
+    "replication_factor": -1, // Replication factor. -1 == all
+    "monitor_ping_interval": "15s" // Time between alive-pings
+  },
+  "consensus": {
+    "raft": { // Raft options, see hashicorp/raft.Configuration
+      "data_folder": "<custom consensus data folder>"
+      "heartbeat_timeout": "1s",
+      "election_timeout": "1s",
+      "commit_timeout": "50ms",
+      "max_append_entries": 64,
+      "trailing_logs": 10240,
+      "snapshot_interval": "2m0s",
+      "snapshot_threshold": 8192,
+      "leader_lease_timeout": "500ms"
+    }
+  },
+  "api": {
+    "restapi": {
+      "listen_multiaddress": "/ip4/127.0.0.1/tcp/9094", // API listen
+      "read_timeout": "30s",
+      "read_header_timeout": "5s",
+      "write_timeout": "1m0s",
+      "idle_timeout": "2m0s",
+      "basic_auth_credentials": [ // leave null for no-basic-auth
+        "user": "pass"
+      ]
+    }
+  },
+  "ipfs_connector": {
+    "ipfshttp": {
+      "proxy_listen_multiaddress": "/ip4/127.0.0.1/tcp/9095", // ipfs-proxy
+      "node_multiaddress": "/ip4/127.0.0.1/tcp/5001", // ipfs-node location
+      "connect_swarms_delay": "7s",
+      "proxy_read_timeout": "10m0s",
+      "proxy_read_header_timeout": "5s",
+      "proxy_write_timeout": "10m0s",
+      "proxy_idle_timeout": "1m0s"
+    }
+  },
+  "monitor": {
+    "monbasic": {
+      "check_interval": "15s" // how often to check for expired metrics/alert
+    }
+  },
+  "informer": {
+    "disk": {
+      "metric_ttl": "30s", // determines how often metric is updated
+      "metric_type": "freespace" // or "reposize"
+    },
+    "numpin": {
+      "metric_ttl": "10s"
+    }
+  }
+}
+```
 
 ## Starting your cluster
 
-`ipfs-cluster-service` will launch your cluster peer. If you have not configured any `cluster_peers` in the configuration, nor any `bootstrap` addresses, a single-peer cluster will be launched.
+`ipfs-cluster-service` will launch your cluster peer. If you have not configured any `cluster/peers` in the configuration, nor any `cluster/bootstrap` addresses, a single-peer cluster will be launched.
```

Comment:
It might be more json-idiomatic to instead write `cluster.peers` and `cluster.bootstrap`

Reply:
changed all occurrences.